backend/de.metas.business/src/main/java/org/adempiere/mm/attributes/api/impl/AttributeSetInstanceBL.java

```diff
@@ -453,9 +453,35 @@ public AttributeSetInstanceId addAttributes(@NonNull final AddAttributesRequest
 		}
 
 		request.getAttributeInstanceBasicInfos().forEach(attributeValue -> {
-			setAttributeInstanceValue(asiId, attributeValue.getAttributeCode(), attributeValue.getValue() );
+			setAttributeInstanceValue(asiId, attributeValue.getAttributeCode(), attributeValue.getValue());
 		});
 
 		return asiId;
 	}
+
+	@Override
+	public boolean asiValuesMatchOrEmpty(@NonNull final AttributeSetInstanceId asiId1, @NonNull final AttributeSetInstanceId asiId2)
+	{
+		final List<I_M_AttributeInstance> attributeInstances1 = attributeDAO.retrieveAttributeInstances(asiId1);
+
+		boolean matches = true;
+
+		for (final I_M_AttributeInstance attributeInstance1 : attributeInstances1)
+		{
+			final I_M_AttributeInstance attributeInstance2 = attributeDAO.retrieveAttributeInstance(asiId2, AttributeId.ofRepoId(attributeInstance1.getM_Attribute_ID()));
```

Comment:
@metas-rc cc @adi-stefan  isn't this an SQL N+1 issue?
why we didn't retrieve all the AIs of second ASI all together?